registrations/test_metrics.py

```diff
@@ -9,14 +9,112 @@
 from datetime import datetime
 from django.contrib.auth.models import User
 from django.test import TestCase
+from django.db.models.signals import post_save
+
+from rest_hooks.models import model_saved
 
 from .metrics import MetricGenerator, send_metric
 from .tests import AuthenticatedAPITestCase
-from .models import Registration, Source
+from .models import (
+        Source, Registration, registration_post_save, fire_created_metric,
+        fire_unique_operator_metric, fire_message_type_metric,
+        fire_receiver_type_metric, fire_source_metric, fire_language_metric,
+        fire_state_metric, fire_role_metric)
 from hellomama_registration import utils
+from changes.models import (Change, change_post_save,
+                            fire_language_change_metric)
 
 
 class MetricsGeneratorTests(AuthenticatedAPITestCase):
+
+    def _replace_post_save_hooks_change(self):
+        def has_listeners():
+            return post_save.has_listeners(Change)
+        assert has_listeners(), (
+            "Change model has no post_save listeners. Make sure"
+            " helpers cleaned up properly in earlier tests.")
+        post_save.disconnect(receiver=change_post_save,
+                             sender=Change)
+        post_save.disconnect(receiver=fire_language_change_metric,
+                             sender=Change)
+        post_save.disconnect(receiver=model_saved,
+                             dispatch_uid='instance-saved-hook')
+        assert not has_listeners(), (
+            "Change model still has post_save listeners. Make sure"
+            " helpers cleaned up properly in earlier tests.")
+
+    def _restore_post_save_hooks_change(self):
+        def has_listeners():
+            return post_save.has_listeners(Change)
+        assert not has_listeners(), (
+            "Change model still has post_save listeners. Make sure"
+            " helpers removed them properly in earlier tests.")
+        post_save.connect(receiver=change_post_save,
+                          sender=Change)
+        post_save.connect(receiver=fire_language_change_metric,
+                          sender=Change)
+        post_save.connect(receiver=model_saved,
+                          dispatch_uid='instance-saved-hook')
+
+    def _replace_post_save_hooks_registration(self):
```

Comment:
If I'm not mistaken, this class inherits `registrations.tests.AuthenticatedAPITestCase` that means this method isn't needed because it is exactly the same as the one here https://github.com/praekelt/hellomama-registration/blob/6a9ffcbf92b06228a51a7626c91ae4a9fbb18a13/registrations/tests.py#L184-L212